ml_metadata/tools/mlmd_bench/fill_types_workload_test.cc

```diff
@@ -23,59 +25,417 @@ limitations under the License.
+  template_config.mutable_fill_types_config()->set_specification(
+      FillTypesConfig::ARTIFACT_TYPE);
+  config_vector.push_back(template_config);
+  template_config.mutable_fill_types_config()->set_specification(
+      FillTypesConfig::EXECUTION_TYPE);
+  config_vector.push_back(template_config);
+  template_config.mutable_fill_types_config()->set_specification(
+      FillTypesConfig::CONTEXT_TYPE);
+  config_vector.push_back(template_config);
+
+  return config_vector;
+}
+
+// Inserts some types into db to set it up in different start status in
+// testing.
+tensorflow::Status InsertTypesInDb(const int64 num_artifact_types,
+                                   const int64 num_execution_types,
+                                   const int64 num_context_types,
+                                   MetadataStore* store) {
+  PutTypesRequest put_request;
+  PutTypesResponse put_response;
+
+  for (int64 i = 0; i < num_artifact_types; i++) {
+    ArtifactType* curr_type = put_request.add_artifact_types();
+    curr_type->set_name(absl::StrCat("pre_insert_artifact_type-", i));
+    (*curr_type->mutable_properties())["property"] = STRING;
+  }
+
+  for (int64 i = 0; i < num_execution_types; i++) {
+    ExecutionType* curr_type = put_request.add_execution_types();
+    curr_type->set_name(absl::StrCat("pre_insert_execution_type-", i));
+    (*curr_type->mutable_properties())["property"] = STRING;
+  }
+
+  for (int64 i = 0; i < num_context_types; i++) {
+    ContextType* curr_type = put_request.add_context_types();
+    curr_type->set_name(absl::StrCat("pre_insert_context_type-", i));
+    (*curr_type->mutable_properties())["property"] = STRING;
+  }
+
+  TF_RETURN_IF_ERROR(store->PutTypes(put_request, &put_response));
+  return tensorflow::Status::OK();
+}
+
+// Executes the given FillTypes workloads.
+tensorflow::Status ExecuteWorkload(MetadataStore* store,
+                                   std::unique_ptr<FillTypes>& workload) {
+  for (int64 i = 0; i < workload->num_operations(); ++i) {
+    OpStats op_stats;
+    TF_RETURN_IF_ERROR(workload->RunOp(i, store, op_stats));
+  }
+  return tensorflow::Status::OK();
+}
+
+// Checks types update status. If the updates are working properly, the type id
+// should remain the same after the updates. On the other hand, the properties
+// size for each type should be greater than before since some new fields have
+// been added in the update process.
+template <typename T>
+bool CheckTypeUpdateStatus(const Type& type_before, const Type& type_after) {
+  if (absl::get<T>(type_before).id() != absl::get<T>(type_after).id()) {
+    return false;
+  }
+  if (absl::get<T>(type_before).properties().size() >=
+      absl::get<T>(type_after).properties().size()) {
+    return false;
+  }
+  return true;
+}
+
+// Checks the first `num_operations` types in `types_before` and `types_after`
+// for their update status.
+bool CheckUpdates(const FillTypesConfig& fill_types_config,
```

Comment:
nit, void 

for the return parse, you can use EXPECT_TRUE / EXPECT_FALSE directly

Reply:
Noted! Fixed.